src/plugins/auto_batch/src/plugin.cpp

```diff
@@ -329,7 +331,58 @@ std::shared_ptr<ov::ICompiledModel> Plugin::compile_model(const std::shared_ptr<
             meta_device.device_batch_size = 1;
         }
     }
-
+    auto ppp_model = model->clone();
+    ov::preprocess::PrePostProcessor preproc(ppp_model);
+    OPENVINO_SUPPRESS_DEPRECATED_START
```

Comment:
Why did you add this code to the auto batch plugin? From my point of view all pre-processing related issues should be already solved on the core level

Reply:
considering a legacy case:
CNNNetwork net;
net.getInputsInfo().begin()->second->setLayout(Layout::NCHW);
net.getInputsInfo().begin()->second->setPrecision(Precision::U8);
ie.LoadNetwork(net, BATCH:GPU, ...)   // load the network with ppp info through legacy API.

in above case, GPU(no matter 1.0 or 2.0) will finally been operated with new layout/precision, but for batch plugin (or other virtual plugin), the model is still with old layout/precision( because in core we will not apply ppp for virtual plugin).
this will result in mismatch of model input/output information between virtual plugin and gpu plugin. for example, the tensors precision in virtual infer request will be differ with the tensors in gpu infer request. that's why here I add the workaround in batch plugin to restore the legacy ppp info to model. 

do  you have other suggestion maybe in core some api I can leverage to handle this situation? @ilyachur

Reply:
Here we have in the core the transformation which adds the legacy preprocessing...
https://github.com/openvinotoolkit/openvino/blob/master/src/inference/src/dev/core_impl.cpp#L804

From my point of view, only this place should add preprocessing to the plugin and inside the plugin we should avoid any WAs.

From the problem description, as I understand the auto batch should have the same issues like the HETERO plugin, but HETERO works without WAs on the plugin side. So maybe we need to take a look how it works and use the same approach for BATCH?

Reply:
em.. from my perspective, hetero vs batch are not exactly same regarding tensor sharing.
for hetero, operate on tensors of infer requests can be fully forwarded to underlying hardware infer requests, which means the  middle layer of hetero infer requests can totally be bypassed.
while on batch (or AUTO), it can allocate its own tensors, and then set these tensors to underlying hardware infer requests during workload scheduling. if there is ppp info mismatch between batch model and hardware model, it can be a problem.

Reply:
But, why auto batch cannot use batch model from the plugin? I mean that generate the model on the plugin side and reuse this information?
Or maybe we can add exception for batch plugin. Anyway I would like to avoid the code duplication here.